src/message.c

```diff
@@ -0,0 +1,88 @@
+#include "message.h"
+
+static channel_group unique_head;
+
+/**
+ * init_message - Initializes dummy head of unique identifier chain
+ */
+void 
+init_message() 
+{
+    unique_head.name = "";
+    INIT_LIST_HEAD(&unique_head.same);
+    INIT_LIST_HEAD(&unique_head.unique);
+    unique_head.push = NULL;
+    unique_head.state = NULL;
+    unique_head.data_size = 0;
```

Comment:
would i replace with a macro

Reply:
Please use functions, where ever possible instead of Macros

Reply:
we provide a function utilizing the macro, but a macro is preferable at this point, since it can be used in non-code segment as static initialization.